tests/unit/mathematics/linalg/Memory_transfer_unittest.cc

```diff
@@ -0,0 +1,80 @@
+#include <shogun/lib/config.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+#ifdef HAVE_VIENNACL
+#include <shogun/mathematics/linalg/LinalgBackendViennaCL.h>
+#endif
+
+#include <shogun/lib/SGVector.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+using namespace linalg;
+
+TEST(LinalgBackendViennaCL, SGVector_to_gpu_viennacl)
+{
+#ifdef HAVE_VIENNACL
+	sg_linalg->set_gpu_backend(new LinalgBackendViennaCL());
+#endif
+
+	const index_t size = 10;
+	SGVector<int32_t> a(size), b(size);
+	a.range_fill(0);
+
+	b = to_gpu(a);
+
+	if (sg_linalg->get_gpu_backend())
+	{
+		EXPECT_EQ(a.on_gpu(), false);
+		EXPECT_EQ(b.on_gpu(), true);
+	}
+	else
+	{
+		EXPECT_EQ(a.on_gpu(), false);
+		EXPECT_EQ(b.on_gpu(), false);
+	}
+}
```

Comment:
Testing memory transfer to GPU should go to the corresponding backend test

Group everything by backend. 
 * one test file for each backend base (generic methods like gpu transfer)
 * one test file for each class of operations under a subfolder of the particular backend

Something like this
```
unit/mathematics/linalg/SGLinalg_unittest.cc
unit/mathematics/linalg/eigen3_backend_unittest.cc // generic tests, no operations
unit/mathematics/linalg/viennacl_backend_unittest.cc // generic tests, no operations, include to_gpu
...
unit/mathematics/linalg/operations/eigen3_operations_unittest.cc // contains dot, etc
unit/mathematics/linalg/operations/viennacl_operations_unittest.cc
```